osquery/tables/system/posix/apt_sources.cpp

```diff
@@ -19,51 +19,95 @@
 namespace osquery {
 namespace tables {
 
-Status parseAptSourceLine(const std::string& line, AptSource& apt_source) {
-  auto comp = osquery::split(line, " ");
-  if (comp.empty() || comp[0][0] == '#') {
-    return Status::failure("Cannot parse comment");
+Status parseAptSourceLine(const std::string& input_line,
+                          AptSource& apt_source) {
+  // attempts to follow conventions in this doc
+  // http://manpages.ubuntu.com/manpages/xenial/man5/sources.list.5.html
+
+  // Remove everything after # from the line for comments
+  auto comment_pos = input_line.find("#");
+  std::string line = input_line;
+  if (comment_pos != std::string::npos) {
```

Comment:
I want to avoid making unused copies of `input_line`, does the following make sense?
```cpp
std::string line;
if (comment_pos != std::string::npos) {
  line = input_line.substr(0, comment_pos);
} else {
  line = input_line;
}
```